### PR TITLE
normy

### PR DESCRIPTION
**Pull request by Torutu**
I got normed

### CODE REVIEW

File: src/parse/redirection_check.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @return 0 if the token is an output redirection token, 1 otherwise.
+ */
+int	handle_redirect_out(char *token, t_token *current_token)
+{
+	if (ft_strncmp(token, ">", 2) == 0 && current_token->in_q == false)
+	{
+		current_token->value = ft_strdup(token);
+		current_token->type = RED_OUT;
+		free(token);
```
Comment (FjjDessoyCaraballo) [inline src/parse/redirection_check.c:29]: Use free_null function please

File: src/parse/redirection_check.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @return 0 if the token is an append redirection token, 1 otherwise.
+ */
+int	handle_append(char *token, t_token *current_token)
+{
+	if (ft_strncmp(token, ">>", 3) == 0 && current_token->in_q == false)
+	{
+		current_token->value = ft_strdup(token);
+		current_token->type = APPEND;
+		free(token);
```
Comment (FjjDessoyCaraballo) [inline src/parse/redirection_check.c:50]: Use free_null please

File: src/parse/redirection_check.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @return 0 if the token is an input redirection token, 1 otherwise.
+ */
+int	handle_redirect_in(char *token, t_token *current_token)
+{
+	if (ft_strncmp(token, "<", 2) == 0 && current_token->in_q == false)
+	{
+		current_token->value = ft_strdup(token);
+		current_token->type = RED_IN;
+		free(token);
```
Comment (FjjDessoyCaraballo) [inline src/parse/redirection_check.c:71]: Use free_null

File: src/parse/redirection_check.c [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * @return 0 if the token is a heredoc redirection token, 1 otherwise.
+ */
+int	handle_heredoc(char *token, t_token *current_token)
+{
+	if (ft_strncmp(token, "<<", 3) == 0 && current_token->in_q == false)
+	{
+		current_token->value = ft_strdup(token);
+		current_token->type = HEREDOC;
+		free(token);
```
Comment (FjjDessoyCaraballo) [inline src/parse/redirection_check.c:92]: Use free_null please

File: src/parse/setup_env.c [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+ * @return A dynamically allocated string containing the extracted word.
+ *
+ * @throws None.
+ */
+char	*extract_word(const char **ptr, int max_word_size)
+{
+	int		index;
+	char	*word;
+
+	word = (char *)malloc((max_word_size + 1) * sizeof(char));
```
Comment (FjjDessoyCaraballo) [inline src/parse/setup_env.c:71]: Malloc check here

**Review by FjjDessoyCaraballo — CHANGES_REQUESTED** [review]

File: src/parse/setup_env.c (hunk 5 above)
Comment (Torutu) [reply src/parse/setup_env.c:71]: I do malloc check it just index = 0; in between

**Pull request merged**